bin/docker_utils.py

```diff
@@ -10,28 +10,40 @@
 
 DOCKER_REGISTRY_API='https://registry-1.docker.io/v2'
 DOCKER_HUB_API='https://hub.docker.com/v2'
-
-def http_request(url, data=None, headers=None, method = 'GET', json=False):
-  response = request(method=method, url=url, data=data, headers=headers)
+DOCKER_HUB_TOKEN = None
+
+def hub_request(uri, data=None, params=None, headers=None, method = 'GET', json=False):
+  global DOCKER_HUB_TOKEN
+  if not DOCKER_HUB_TOKEN:
+    DOCKER_HUB_TOKEN = get_token()
+  if not headers: headers = {}
+  headers['Authorization'] = 'JWT %s' % DOCKER_HUB_TOKEN
+  return http_request('%s%s' %(DOCKER_HUB_API, uri), data, params, headers, method, json)
+
+def http_request(url, data=None, params=None, headers=None, method = 'GET', json=False):
+  response = request(method=method, url=url, data=data,  params=params, headers=headers)
   if not response.ok:
     return response
   return response.json() if json else response
 
 # get token for docker Registry API
-def get_docker_token(repo):
-  return http_request('https://auth.docker.io/token?service=registry.docker.io\
-&scope=repository:%s:pull' % repo, json=True)['token']
+def get_registry_token(repo):
+  uri = 'https://auth.docker.io/token'
+  payload = {
+    'service' : 'registry.docker.io',
+    'scope' : 'repository:%s:pull' % repo
+  }
+  return http_request(uri, params = payload, json = True)['token']
 
 # get token for Docker HUB API:
-def get_dockerHubToken(DOCKER_HUB_API):
-  url = '%s/users/login/' % DOCKER_HUB_API
+def get_token():
+  uri = '%s/users/login/' % DOCKER_HUB_API
   secret = open(expanduser("~/.docker-token")).read().strip()
-  response = http_request(url, loads(secret), method = 'POST', json=True)
-  return response['token']
+  return http_request(uri, loads(secret), method = 'POST', json=True)['token']
 
-def get_repos(DOCKER_HUB_API, user):
-  url = '%s/repositories/%s/' % (DOCKER_HUB_API, user)
-  response = http_request(url, json=True)
+def get_repos(user):
+  uri = '/repositories/%s/' % user
```

Comment:
@tulamor , one think I noticed that it does not return all the repositories. You either need to also send `page_size` in the url e.g. `url=/repositories/%s/?page_size=1000` to get 1000 repositories in the response or recursively call the docker hub api if you find `next` key in the response. For example in normal response you will get some thing like
```
'{"count": 75,
  "next": "https://hub.docker.com/v2/repositories/cmssw/?page=2",
  "previous": null, 
  "results": [{....}]
}
```
so yuo can recursively call the api if `next` is not `null` and accomulate results.

Reply:
fixed with page_size parameter